src/matlab2tikz.m

```diff
@@ -2475,17 +2475,16 @@ function assertRegularAxes(handle)
 end
 % ==============================================================================
 function [m2t, str] = drawHggroup(m2t, h)
-% Octave doesn't have the handle() function, so there's no way to determine
-% the nature of the plot anymore at this point.  Set to 'unknown' to force
-% fallback handling. This produces something for bar plots, for example.
+% Continue according to the plot type. Since the function `handle` is
+% not available in Octave, the plot type will be guessed or the fallback type
+% 'unknown' used.
 % #COMPLEX: big switch-case
-% TODO: Introduce function `guessOctaveType()` to infer plot type by trying
-% to read certain properties, that are specific to a plot type (for example
-% `udata` should be unique to errorbarseries). See #645 for more details.
-    try
+    if strcmpi(getEnvironment, 'MATLAB')
         cl = class(handle(h));
-    catch %#ok
-        cl = 'unknown';
+    elseif strcmpi(getEnvironment, 'octave')
```

Comment:
Hmm, only nit-pick, maybe change this to:
```
switch getEnvironment()
    case 'MATLAB'
       ...
    case 'Octave'
       ...
    otherwise
       errorUnknownEnvironment()
end
```
?